test/clientconn_state_transition_test.go

```diff
@@ -164,7 +164,7 @@ func testStateTransitionSingleAddress(t *testing.T, want []connectivity.State, s
 		connMu.Unlock()
 	}()
 
-	client, err := grpc.Dial("",
+	client, err := grpc.NewClient("passthrough:///",
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
 		grpc.WithDefaultServiceConfig(fmt.Sprintf(`{"loadBalancingConfig": [{"%s":{}}]}`, stateRecordingBalancerName)),
 		grpc.WithDialer(pl.Dialer()),
```

Comment:
After line 182, you need to wait for the test balancer to be built before getting the state notification channel. The balancer is built when the channel exits Idle. You can add the following lines:
```go
go testutils.StayConnected(ctx, client)

// Wait for the test balancer to be built before capturing it's state
// notification channel.
testutils.AwaitNotState(ctx, t, client, connectivity.Idle)
```

Reply:
done